manager/src/main/java/io/gingersnapproject/Caches.java

```diff
@@ -60,6 +65,26 @@ private void actualRemove(String name, String key, Uni<String> value) {
       }
    }
 
+   public  Uni<String> denormalizedPut(String name, String key, String value) {
+      Table table = databaseHandler.table(configuration.rules().get(name).connector().table());
+      if (!table.foreignKeys().isEmpty()) {
+         // We have foreign keys to resolve
+         Json json = Json.read(value);
+         UniJoin.Builder<Object> builder = Uni.join().builder();
+         for (ForeignKey fk : table.foreignKeys()) {
+            // TODO: handle composite fks
+            Json fkJson = json.atDel(fk.columns().get(0));
```

Comment:
The `fkJson` could be null if the user does a projection in db-syncer.

Reply:
added a `null` check

Reply:
Yeah, the more I am thinking about this. We really need to shift to key and value columns for eager mode. Then we can verify this stuff at startup of cache manager and throw an error at runtime if the db-syncer doesn't have the same columns.